### ISSUE TITLE
Categorize Hyper-parameters.

### ISSUE THREAD

**Issue by bwanglzu** [faneshion/MatchZoo#137]
Check all the hyperparameters in varies models, , categorize them into common parameters or model specific parameters

**Comment by bwanglzu** [comment]
CDSSM (CLSM)
```python
w_initializer = 'glorot_normal'
b_initializer = 'zeros'
dim_hidden = 300
dim_fan_out = 128
dim_trigram = X # dimension of tri-letter, based on vocab size
window_size = 3  # 3-gram
padding = 'same'
act_hidden = 'tanh'
num_epochs = 20
batch_size = 1024
train_test_split = 0.2
verbose = 1
shuffle = True
lr = 0.1
decay = 1e-6
strides = 1
neg_count = 4 # sample 4 negative documents
metrics=['accuracy']
optimizer=SGD
reg_rate=0.2
dropout_rate=0.0
pool_size=2
act_prediction = 'softmax'
```

**Comment by faneshion** [comment]
DRMM
w_initializer = 'glorot_normal'
b_initializer = 'zeros'
dim_hidden_layers = [ 30, 5] 
act_hidden = 'tanh'
num_epochs = 20
batch_size = 1024
train_test_split = 0.2 
verbose = 1
shuffle = True
lr = 0.1
decay = 1e-6
strides = 1
neg_count = 4 # sample 4 negative documents
metrics=['accuracy']
optimizer=SGD
reg_rate=0.2
dropout_rate=0.0
pool_size=2
sim='cos' # cosine similarity

**Comment by faneshion** [comment]
ARC-I @bwanglzu I guess the strides is for learning rate, am I right?

w_initializer = 'glorot_normal'
b_initializer = 'zeros'
num_kernel = 128 # num of kernel in 1D convolution
kernel_size = 3  # 1D convolution size
pool_size_1 = 3 # 1D pooling size for sentence 1
pool_size_2 = 3 # 1D pooling size for sentence 2
dim_hidden_layers = [30, 10] # full connected layer in the final interaction layers.
dropout_rate = 0.5 
act_hidden = 'tanh'
num_epochs = 20
batch_size = 1024
train_test_split = 0.2
verbose = 1
shuffle = True
lr = 0.1
decay = 1e-6
strides = 1
neg_count = 4 # sample 4 negative documents
metrics=['accuracy']
optimizer=SGD
reg_rate=0.2
dropout_rate=0.0
pool_size=2
sim='cos' # cosine similarity

**Comment by faneshion** [comment]
ARC-II

w_initializer = 'glorot_normal'
b_initializer = 'zeros'
1d_num_kernel = 128 # num of kernel in 1D convolution
1d_kernel_size = 3 # 1D convolution size
num_conv2d_layers = 2 # number of 2D convolution layers
2d_kernel_counts = [32, 32] # number of kernel in 2D convolution layers.
2d_kernel_sizes = [[3, 3], [3, 3]] # a list of 2D kernel size where the number is determined by 2d_num_layers.
2d_mpool_sizes = [[3, 3], [3, 3]]  # a list of 2D max pooling size where the number is determined by 2d_num_layers.
dropout_rate = 0.5
act_hidden = 'tanh'
num_epochs = 20
batch_size = 1024
train_test_split = 0.2
verbose = 1
shuffle = True
lr = 0.1
decay = 1e-6
strides = 1
neg_count = 4 # sample 4 negative documents
metrics=['accuracy']
optimizer=SGD
reg_rate=0.2
dropout_rate=0.0
pool_size=2
sim='cos' # cosine similarity

**Comment by bwanglzu** [comment]
@faneshion thanks for the input, here `stride` denotes for **step size of CNN filters**, according to [cs231n](http://cs231n.github.io/convolutional-networks/):

>  we must specify the stride with which we slide the filter. When the stride is 1 then we move the filters one pixel at a time. When the stride is 2 (or uncommonly 3 or more, though this is rare in practice) then the filters jump 2 pixels at a time as we slide them around. This will produce smaller output volumes spatially.
